hs_geo_raster_resource/tests/test_raster_metadata.py

```diff
@@ -0,0 +1,569 @@
+        text_file = open(temp_text_file, 'w')
+        text_file.write("Raster records")
+        self.text_file_obj = open(temp_text_file, 'r')
+
+    def tearDown(self):
+        super(TestRasterMetaData, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_allowed_file_types(self):
+        # test allowed file type is '.tif, .zip'
+        self.assertIn('.tif', RasterResource.get_supported_upload_file_types())
+        self.assertIn('.zip', RasterResource.get_supported_upload_file_types())
+        self.assertEquals(len(RasterResource.get_supported_upload_file_types()), 2)
+
+        # there should not be any content file
+        self.assertEquals(self.resRaster.files.all().count(), 0)
+
+        # trying to add a text file to this resource should raise exception
+        files = [UploadedFile(file=self.text_file_obj, name=self.text_file_obj.name)]
+        with self.assertRaises(utils.ResourceFileValidationException):
+            utils.resource_file_add_pre_process(resource=self.resRaster, files=files, user=self.user,
+                                                extract_metadata=False)
+
+        # trying to add bad .tif file should raise file validation error
+        files = [UploadedFile(file=self.raster_bad_tif_file_obj, name=self.raster_bad_tif_file_name)]
+        with self.assertRaises(utils.ResourceFileValidationException):
+            utils.resource_file_add_pre_process(resource=self.resRaster, files=files, user=self.user,
+                                                extract_metadata=False)
+
+        # trying to add bad .zip file should raise file validation error
+        files = [UploadedFile(file=self.raster_bad_zip_file_obj, name=self.raster_bad_zip_file_name)]
+        with self.assertRaises(utils.ResourceFileValidationException):
+            utils.resource_file_add_pre_process(resource=self.resRaster, files=files, user=self.user,
+                                                extract_metadata=False)
+
+        # trying to add good .tif file should pass the file check
+        files = [UploadedFile(file=self.raster_tif_file_obj, name=self.raster_tif_file_name)]
+        utils.resource_file_add_pre_process(resource=self.resRaster, files=files, user=self.user,
+                                            extract_metadata=False)
+        utils.resource_file_add_process(resource=self.resRaster, files=files, user=self.user,
+                                        extract_metadata=False)
+
+        # there should be 2 content file: with .vrt file created by system
+        self.assertEquals(self.resRaster.files.all().count(), 2)
+        file_names = [os.path.basename(f.resource_file.name) for f in self.resRaster.files.all()]
+        self.assertIn('raster_tif_valid.vrt', file_names)
+
+        # delete content file that we added above
+        hydroshare.delete_resource_file(self.resRaster.short_id, self.raster_tif_file_name, self.user)
+
```

Comment:
How many content files the resource should have at this point? Should test that before adding the zip file.